connectors/src/connectors/notion/temporal/activities.ts

```diff
@@ -261,31 +268,33 @@ export async function getPagesAndDatabasesToSync({
       skippedDatabaseIds
     );
   } catch (e) {
-    // Sometimes a cursor will consistently fail with 500.
-    // In this case, there is not much we can do, so we just give up and move on.
-    // Notion workspaces are resynced daily so nothing is lost forever.
-    const potentialNotionError = e as {
-      body: unknown;
-      code: string;
-      status: number;
-    };
-    if (
-      potentialNotionError.code === "internal_server_error" &&
-      potentialNotionError.status === 500
-    ) {
-      if (Context.current().info.attempt > 20) {
-        localLogger.error(
-          {
-            error: potentialNotionError,
-            attempt: Context.current().info.attempt,
-          },
-          "Failed to get Notion search result page with cursor. Giving up and moving on"
-        );
-        return {
-          pageIds: [],
-          databaseIds: [],
-          nextCursor: null,
-        };
+    if (e instanceof Error && isNotionError(e)) {
```

Comment:
Made sure that NotionHQ.Client() uses `Error` in their base errors.

https://github.com/makenotion/notion-sdk-js/blob/ede416af263e8a866a19b912bcd022369e8009a9/src/errors.ts#L38-L42

Reply:
Do they all have `code` ? is `code` really a good probe for this?

Maybe we should test that code is either "internal_server_error" | "unauthorized" in isNotionError?